_posts/news/2017-11-29-announcing-scalajs-1.0.0-M2.md

```diff
@@ -0,0 +1,435 @@
+This makes it more consistent with other tasks such as `run` and `test`.
+
+### `scalajs-javalib-ex` was removed
+
+The artifact `scalajs-javalib-ex` is removed in 1.x.
+It only contained a partial implementation of `java.util.ZipInputStream`.
+If you were using it, we recommend that you integrate a copy of [its source code from Scala.js 0.6.x](https://github.com/scala-js/scala-js/tree/0.6.x/javalib-ex/src/main/scala/java/util/zip) into your project.
+
+### `js.use(x).as[T]` was removed
+
+The use cases for `js.use(x).as[T]` have been dramatically reduced by non-native JS classes (previously known as Scala.js-defined JS classes).
+This API seems virtually unused on the visible Web.
+Moreover, it was the only macro in the Scala.js standard library.
+
+We have therefore removed it from the standard library, and it is not provided anymore.
+On demand, we can republish it as a separate library, if you need it.
+
+### Scala 2.12.0 is not supported anymore
+
+A severe regression in Scala 2.12.0 upstream, affecting `js.UndefOr`, forced us to drop support for Scala 2.12.0.
+Scala 2.12.1+ is supported.
+
+## Enhancements
+
+There are very few enhancements in Scala.js 1.0.0-M2.
+Scala.js 1.0.0 is focused on simplifying Scala.js, not on adding new features.
+Nevertheless, here are a few enhancements.
+
+### Scala.js can access `require` and other magical "global" variables of special JS environments
+
+The changes from global *object* to global *scope* mean that magical "global" variables provided by some JavaScript environments, such as `require` in Node.js, are now visible to Scala.js.
+For example, it is possible to dynamically call `require` as follows in Scala.js 1.x:
+
+{% highlight scala %}
+val pathToSomeAsset = "assets/logo.png"
+val someAsset = js.Dynamic.global.require(pathToSomeAsset)
+{% endhighlight %}
+
+We still recommend to use `@JSImport` and `CommonJSModule` for statically known imports.
+
+### Declaring inner classes in native JS classes
+
+Some JavaScript APIs define classes inside objects, as in the following example:
+
+{% highlight javascript %}
+class OuterClass {
+  constructor(x) {
+    this.InnerClass = class InnerClass {
+      someMethod() {
+        return x;
+      }
+    }
+  }
+}
+{% endhighlight %}
+
+allowing use sites to instantiate them as
+
+{% highlight javascript %}
+const outerObject = new OuterClass(42);
+const innerObject = new outerObject.InnerClass();
+console.log(innerObject.someMethod()); // prints 42
+{% endhighlight %}
+
+In Scala.js 0.6.x, it is very awkward to define a facade type for `OuterClass`, as illustrated [in issue #2398](https://github.com/scala-js/scala-js/issues/2398).
+Scala.js 1.x now allows to declare them very easily as inner JS classes:
+
+{% highlight scala %}
+@js.native
+@JSGlobal
+class OuterClass(x: Int) extends js.Object {
+  @js.native
+  class InnerClass {
```

Comment:
`extends js.Object`, not `@js.native`?

Reply:
`extends js.Object` is missing indeed. `@js.native` is necessary. See [the test here](https://github.com/scala-js/scala-js/blob/617c4170fe9844eea0115b6ed59b2816201a70e7/test-suite/js/src/test/scala/org/scalajs/testsuite/compiler/InteroperabilityTest.scala#L783)

Reply:
Ah, right. The outer class is native as well. OK :)